api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java

```diff
@@ -452,9 +462,85 @@ public void validate_shouldPassIfDifferentConceptsHaveTheSameShortNames() throws
+	public void validate_shouldFailIfAnyDescriptionIsNotEnteredWhileCreatingANewConcept() throws Exception {
+		Concept concept = new Concept();
+		concept.addName(new ConceptName("some name", Context.getLocale()));
+		Errors errors = new BindException(concept, "concept");
+		new ConceptValidator().validate(concept, errors);
+		Assert.assertTrue(errors.hasFieldErrors("descriptions"));
+	}
+	
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "fail if any description is a removed while editing a new concept ", method = "validate(Object,Errors)")
+	public void validate_shouldFailIfAnyDescriptionIsRemovedWhileEditingAExistingConcept() throws Exception {
+		Concept concept = Context.getConceptService().getConcept(5497);
+		ConceptDescription description = concept.getDescription();
+		concept.removeDescription(description);
+		Errors errors = new BindException(concept, "concept");
+		new ConceptValidator().validate(concept, errors);
+		Assert.assertTrue(errors.hasFieldErrors("descriptions"));
+	}
+	
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "should pass if  none of the concept descriptions is null", method = "validate(Object,Errors)")
+	public void validate_shouldPassIfNoneofTheConceptDescriptionsIsNull() throws Exception {
+		Concept concept = new Concept();
+		List<Locale> locales = Context.getAdministrationService().getAllowedLocales();
+		for (Locale loc : locales) {
+			concept.addName(new ConceptName("some name", loc));
+			concept.addDescription(new ConceptDescription("some description", loc));
+		}
+		Errors errors = new BindException(concept, "concept");
+		new ConceptValidator().validate(concept, errors);
+		Assert.assertFalse(errors.hasErrors());
+	}
+	
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "should fail if  atleast one ConceptDescription is not found in Concept", method = "validate(Object,Errors)")
+	public void validate_shouldFailIfAtleastOneDescriptionIsNotFound() throws Exception {
+		Concept concept = new Concept();
+		List<Locale> locales = Context.getAdministrationService().getAllowedLocales();
+		for (Locale loc : locales) {
+			concept.addDescription(new ConceptDescription(null, loc));
+		}
+		Errors errors = new BindException(concept, "concept");
+		new ConceptValidator().validate(concept, errors);
+		Assert.assertTrue(errors.hasErrors());
+	}
+	
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "should Not return a Null Pointer Exception", method = "validate(Object,Errors)")
+	public void validate_shouldNotReturnaNullPointerException() throws Exception {
+		Concept concept = new Concept();
+		Collection<ConceptDescription> descriptions = concept.getDescriptions();
+		Assert.assertTrue(descriptions.isEmpty());
+		Assert.assertNotNull(descriptions.size());

```

Comment:
Yes i want to see that test as part of your pull request.

On Thu, Feb 26, 2015 at 8:34 PM, Sandeep Raparthi <notifications@github.com>
wrote:

> In api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java
> <https://github.com/openmrs/openmrs-core/pull/1354#discussion_r25446489>:
>
> > +		}
> > +		Errors errors = new BindException(concept, "concept");
> > +		new ConceptValidator().validate(concept, errors);
> > +		Assert.assertTrue(errors.hasErrors());
> > +	}
> > +	
> > +	/**
> > +	 * @see ConceptValidator#validate(Object,Errors)
> > +	 **/
> > +	@Test
> > +	@Verifies(value = "should Not return a Null Pointer Exception", method = "validate(Object,Errors)")
> > +	public void validate_shouldNotReturnaNullPointerException() throws Exception {
> > +		Concept concept = new Concept();
> > +		Collection<ConceptDescription> descriptions = concept.getDescriptions();
> > +		Assert.assertTrue(descriptions.isEmpty());
> > +		Assert.assertNotNull(descriptions.size());
>
> @dkayiwa <https://github.com/dkayiwa> shall i make that change and update
> the PR ? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1354/files#r25446489>.
>



-- 
If we keep uppermost in our minds the unkind and unjust acts of others, we
shall find it impossible to love them as Christ has loved us; but if our
thoughts dwell upon the wondrous love and pity of Christ for us, the same
spirit will flow out to others.